src/Vegetation/Vegetation.jl

```diff
@@ -0,0 +1,511 @@
+prognostic_types(model::VegetationModel{FT}) where {FT} = (FT,)
+
+"""
+    flux(
+        z1::FT,
+        z2::FT, 
+        p1::FT, 
+        p2::FT, 
+        a1::FT, 
+        a2::FT, 
+        b1::FT, 
+        b2::FT, 
+        K_sat1::FT, 
+        K_sat2::FT) where {FT} 
+Computes the water flux given 1) the absolute pressure at two points located
+at z1 and z2, corresponding here to the middle of each compartment 
+(for eg. a stem or leaf compartment), and 2) the maximum conductivity along 
+the flow path between these two points, assuming an arithmetic 
+mean for mean K_sat between the two points (Bonan, 2019; Zhu, 2008) 
+to take into account the change in K_sat halfway 
+betwwen z1 and z2. The expression is given in full in the Clima docs
+in section 6.4 "Bulk plant hydraulics model".
+"""
+function flux(
+    z1::FT,
+    z2::FT,
+    p1::FT,
+    p2::FT,
+    a1::FT,
+    a2::FT,
+    b1::FT,
+    b2::FT,
+    K_sat1::FT,
+    K_sat2::FT,
+) where {FT}
+    u1 = a1 * exp(b1 * p1)
+    u2 = a1 * exp(b1 * p2)
+    num1 = log(u1 + FT(1))
+    num2 = log(u2 + FT(1))
+    c1 = K_sat1 * (a1 + FT(1)) / a1
+    term1 = -c1 / b1 * (num2 - num1) / (z2 - z1)
+
+    c2 = K_sat2 * (a2 + FT(1)) / a2
+    term2_up = c2 * (FT(1) - FT(1) / (FT(1) + a2 * exp(b2 * p2)))
+    term2_do = c1 * (FT(1) - FT(1) / (FT(1) + a1 * exp(b1 * p1)))
+    term2 = -(term2_up - term2_do) / 2
+    flux = term1 + term2
+    return flux # units of [m s-1]
+end
+
+"""
+    augmented_liquid_fraction(
+        ν::FT, 
+        S_l::FT) where {FT}
+Computes the augmented liquid fraction from porosity and
+effective saturation. Augmented liquid fraction allows for
+oversaturation: an expansion of the volume of space
+available for storage in a plant compartment.
+"""
+function augmented_liquid_fraction(ν::FT, S_l::FT) where {FT}
+    ϑ_l = S_l * ν
+    safe_ϑ_l = max(ϑ_l, 0)
+    return safe_ϑ_l # units of [m3 m-3]
+end
+
+"""
+    effective_saturation_vegetation(
+        ν::FT, 
+        ϑ_l::FT) where {FT}
+Computes the effective saturation (volumetric water content relative to
+porosity).
+"""
+function effective_saturation_vegetation(ν::FT, ϑ_l::FT) where {FT}
+    S_l = ϑ_l / ν # S_l can be > 1
+    safe_S_l = max(S_l, 0)
+    return safe_S_l # units of [m3 m-3]
+end
+
+"""
+    van_Genuchten_volume_to_pressure(
+        α::FT, 
+        n::FT, 
+        m::FT, 
+        S_l::FT) where {FT}
+Converts effective saturation to absolute pressure, in unsaturated case.
+"""
+function van_Genuchten_volume_to_pressure(
+    α::FT,
+    n::FT,
+    m::FT,
+    S_l::FT,
+) where {FT}
+    p = -((S_l^(-FT(1) / m) - FT(1)) * α^(-n))^(FT(1) / n)
+    return p # units of [m]
+end
+
+"""
+    van_Genuchten_pressure_to_volume(
+        α::FT, 
+        n::FT, 
+        m::FT, 
+        p::FT) where {FT}
+Converts absolute pressure to effective saturation, in unsaturated case.
+"""
+function van_Genuchten_pressure_to_volume(α::FT, n::FT, m::FT, p::FT) where {FT}
```

Comment:
this does not return volume... (nor does the inverse take volume). better naming?

Reply:
True, I guess I had named it like this because Christoffersen had and I thought it was a common name but having searched elsewhere in my papers, I do not see this being used.  More specific ones which Ive seen elsewhere inclue : water-retention curve, moisture characteristic, or water release curve. We could also go for p_to_var_theta; but I think I perfer water_retention_curve, any thoughts?

Reply:
hmm actually water retention doesnt work as I need the inverse function so maybe more p_to_var_theta and var_theta_to_p

Reply:
`water_retention_curve` and `inverse_water_retention_curve`?

Reply:
yes, that works :)